turn/turn_rest.go

```diff
@@ -0,0 +1,75 @@
+package turn
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/rs/xid"
+	"github.com/screego/server/config"
+)
+
+type TurnREST struct {
+	ttl    time.Duration
+	secret []byte
+	port   int
+}
+
+func (t *TurnREST) AcceptAccounts(client, host *TurnAccount) error {
+	err := buildPassword(client, t.ttl, t.secret)
+	if err != nil {
+		return err
+	}
+	return buildPassword(host, t.ttl, t.secret)
+}
+
+func buildPassword(account *TurnAccount, ttl time.Duration, secret []byte) error {
+	//https://stackoverflow.com/questions/35766382/coturn-how-to-use-turn-rest-api#54725092
+	if ttl <= 0 {
+		return errors.New("Use a TTL > 0")
+	}
+	account.Username = fmt.Sprintf("%d:%s", time.Now().Add(ttl).Unix(), account.Id.String())
+	mac := hmac.New(sha1.New, secret)
+	_, err := mac.Write([]byte(account.Username))
+	if err != nil {
+		return err
+	}
+	account.Credential = base64.StdEncoding.EncodeToString(mac.Sum(nil))
+	return nil
+}
+
+func (t *TurnREST) RevokeAccounts(...xid.ID) {
+	// do nothing, wait for peremption
+}
+
+func (t *TurnREST) Port() int {
+	return t.port
+}
+
+func newTurnREST(conf config.Config) (TurnServer, error) {
+	ttl, err := time.ParseDuration(conf.TurnCoturnSignedTtl)
+	if err != nil {
+		return nil, err
+	}
+	if conf.TurnCoturnSignedSecret == "" {
+		return nil, errors.New("TurnCoturnSignedSecret can't be empty")
+	}
+	slugs := strings.Split(conf.TurnAddress, ":")
+	if len(slugs) == 0 {
+		return nil, fmt.Errorf("Can't read CoturnAddress : %s", conf.TurnAddress)
+	}
+	port, err := strconv.Atoi(slugs[len(slugs)-1])
+	if err != nil {
+		return nil, fmt.Errorf("Can't parse TurnAddress port : %s %s", slugs[len(slugs)-1], err)
+	}
```

Comment:
It's a little unfortunate that we need to specify addresses pointing to TURN server (IPv4 and IPv6) in `ExternalIP` parameter (which is used in `ws/room.go` in `addresses()`, and port to the TURN server in `TurnAddress`, where the IP address of `TurnAddress` is practically useless...

Not sure what you guys think about it, is it just a matter of properly documenting this, or can you think of any other way to configure this?

Maybe `TurnAddress` can be parsed in a similar way as `ExternalIP` (i.e. it can take both IPv4 and IPv6 values), and then add `IpAddressV4` and `IpAddressV6` to `TurnServer` struct, so that in `ws/room.go::addresses()` we would use like below?

```go
fmt.Sprintf("%s:%s:%d?transport=tcp", prefix, r.turnServer.IpAddressV4(), r.turnServer.Port()))
```

Reply:
OK, it take time, I'm bit tired, but I understand your point.

`config.Config.ExternalIP` is only used for the turn server (without port), and `config.Config.TurnAddress` is a full address but only use the port part.

`config.Config.ServerAddress` is used for the web service (ip+port).